apps/mobile/src/components/Camera/QrCamera.tsx

```diff
@@ -0,0 +1,202 @@
+                    }}
+                  >
+                    <Badge
+                      themeName="badge_background"
+                      circleSize="$9"
+                      content={<SafeFontIcon size={20} name="close" />}
+                    />
+                  </Pressable>
+                </View>
+
+                <View flex={1} justifyContent={'flex-end'} alignItems={'center'} marginBottom={'$8'}>
+                  {typeof heading === 'string' ? <H3>{heading}</H3> : heading}
+                </View>
+              </View>
+            </BlurView>
+
+            {/* Middle with transparent center */}
+            <View style={styles.transparentCenter}>
+              <BlurView
+                style={[styles.sideBlur, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+                intensity={30}
+                // experimentalBlurMethod={'dimezisBlurView'}
+                tint={'systemUltraThinMaterialDark'}
+              />
+
+              <View style={[styles.transparentBox, denied && { backgroundColor: 'rgba(0, 0, 0, 0.8)' }]}>
+                {/* Green corners */}
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topLeft]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topRight]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomLeft]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomRight]} />
+
+                {denied && (
+                  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+                    <SafeFontIcon name={'camera'} size={40} color={'$error'} />
+                    <SafeButton rounded secondary onPress={openSettings} marginTop={20}>
+                      Enable camera
+                    </SafeButton>
+                  </View>
+                )}
+              </View>
+              <BlurView
+                style={[styles.sideBlur, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+                intensity={30}
+                // experimentalBlurMethod={'dimezisBlurView'}
+                tint={'systemUltraThinMaterialDark'}
+              />
+            </View>
+
+            {/* Bottom Blur */}
+            <BlurView
+              style={[styles.blur, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+              intensity={30}
+              // experimentalBlurMethod={'dimezisBlurView'}
+              tint={'systemUltraThinMaterialDark'}
+            >
+              <View style={styles.text} paddingVertical={'$8'}>
+                {footer}
+              </View>
+            </BlurView>
+          </View>
+        </View>
+      </View>
+    </Theme>
+  )
+}
+
+const BOX_RADIUS = 5 // Rounded corners
+const CORNER_SIZE = 30 // Size of the green corners
+
+const styles = StyleSheet.create({
```

Comment:
can we move these styles to a file called `styled.ts` and use it from there? Also might be better to style the component in the way as it is suggested from tamagui? You can check how we are doing in the `SafeInput/styled.ts`